x-pack/legacy/plugins/ml/public/application/util/url_state.ts

```diff
@@ -18,13 +18,18 @@ import { getNestedProperty } from './object_utils';
 export type SetUrlState = (attribute: string | Dictionary<any>, value?: any) => void;
 export type UrlState = [Dictionary<any>, SetUrlState];
 
+const decodedParams = new Set(['_a', '_g']);
 function getUrlState(search: string) {
   const urlState: Dictionary<any> = {};
   const parsedQueryString = queryString.parse(search);
 
   try {
     Object.keys(parsedQueryString).forEach(a => {
-      urlState[a] = decode(parsedQueryString[a]) as Dictionary<any>;
+      if (decodedParams.has(a)) {
+        urlState[a] = decode(parsedQueryString[a]) as Dictionary<any>;
+      } else {
+        urlState[a] = parsedQueryString[a];
```

Comment:
i don't know if items that aren't `_a` or `_g` are needed at all for the url state.
but it might also do no harm collecting them. @walterra would need to confirm

Reply:
At the moment also setting the `urlState` expects all of url state so I suggest to keep it like that for now. I did it this way because `useLocation` is not mutable and I always needed the full most recent value for getting/setting `search`.

@darnautov Thanks for adding the test! Seems you forgot to commit to export `getUrlState` so the test fails on CI at the moment. Can you also add a test to check the other way around when setting the url state? I suspect we also need a check against `decodedParams` when setting the state [here](https://github.com/elastic/kibana/blob/24181759bc881be5f8a06bcec309e6fe0347fba7/x-pack/legacy/plugins/ml/public/application/util/url_state.ts#L78).

Reply:
Indeed I forgot to include export to my initial commit, added in https://github.com/elastic/kibana/pull/54915/commits/40d9646682e48e1f699c04874bed9076b1d66203. 
test for `setUrlState`: https://github.com/elastic/kibana/pull/54915/commits/177baeb98a26347514da87b11ea43d8792e0ce1a